src/models/BrowserStyleSheet.js

```diff
@@ -331,6 +331,10 @@ if (!DISABLE_SPEEDY) {
 
       comp.textNode.appendData(css.join(' '))
 
+      if (comp.textNode.parentNode !== this.el) {
```

Comment:
This fixes an issue with styles injected after initial mount where `css` was not injected at all due to the saved `textNode` not being live in the document.

Reply:
Wait, this should never be necessary. What's going on here in detail? This should inject the component's textNode:

https://github.com/marionebl/styled-components/blob/133afe8591434dcdae8b834bdfe2a69d449d2a12/src/models/BrowserStyleSheet.js#L313

Reply:
OK, will have a detailed look at what goes wring in `addComponent`

Reply:
Ok, my extended use case has some unrelated script editing the underlying `style` tag between `addComponent` and `injectStyles`, leading to the reference of some `this.components` values pointing to `null`. 

I understand this is not necessary looking only at the internals of `styled-components`, but this might be a sensible safety measure (plus it enables my use case :)). What do you think?

Reply:
Alright, just be aware that different conditions apply to insertRule, but it’s a good safety measure.

Can you add a comment please? I think it’d be great to make clear that this will be dead code in a standard environment